app/controllers/fhir/r4/api_controller.rb

```diff
@@ -94,10 +101,18 @@ def update
         :'system/Patient.*'
       )
 
-      updates = Patient.from_fhir(contents, default_patient_jurisdiction_id)
-
       resource = get_patient(params.permit(:id)[:id])
 
+      if request.patch? && !resource.nil?
+        begin
+          contents = apply_patch(resource, patch)
+        rescue StandardError => e
+          status_bad_request([['Unable to apply patch', e&.message].compact.join(': ')]) && return
```

Comment:
I found a case where we don't catch an error that we probably should. If I put an invalid `"path"`, it seems to just silently ignore it. For example, if I request:
```json
[
  { "op": "add", "path": "/birthDate", "value": "1985-03-25" },
  { "op": "add", "path": "/birthDat", "value": "1985-03-26" }
]
```
The result is that the patient's `birthDate` is set to `"1985-03-25"`. This seems in conflict with this from the patch RFC (emphasis my own):
>If a normative requirement is violated by a JSON Patch document, **or
   if an operation is not successful**, evaluation of the JSON Patch
   document SHOULD terminate and application of the entire patch
   document SHALL NOT be deemed successful.

So my understanding is that this whole patch should fail, since the second operation fails. I realize we're pretty limited by the gem here, but do you think this is a case we could catch? Do you think my interpretation of the RFC is too strict? I could maybe see an argument that the second operation didn't _fail_ per se, but rather just had no effect since it uses a non-existent path. But RFC rules aside, I think this would be a helpful for clients if we catch this kind of thing.

Reply:
Good point, this is a tricky one though. I've stepped through the code, and what I'm seeing is that the patch is actually successfully applied to the JSON representation of the resource. But then when we run `FHIR.from_contents`, the `birthDat` field is ignored. It seems like on POST and PUT, similar things happen. You can add as many non-existent fields as you want, and no errors occur, the fields are just ignored.

I'm trying to think of a reasonable approach for rejecting unrecognized fields that wouldn't require us to duplicate a lot of what the `fhir_models` gem is doing for us, but I'm not sure there is an easy solution.

Reply:
Yeah, fair enough. I'm ok leaving this as is.